### DISCUSSION TITLE
How to maintain jp2 settings

### DISCUSSION THREAD

**Discussion by Jmuccigr** [ImageMagick/ImageMagick#2569] [Help]
Unusually I have a jp2 file to work with. When I do something as simple as change its density, it increases massively in size. Is there a simple way to read and then rewrite the relevant settings, so this doesn't happen?

(I'm also curious about why this happens.)

TIA.

MacOS 10.13.6
Version: ImageMagick 7.0.10-29 Q16 x86_64 2020-09-07 https://imagemagick.org
Copyright: © 1999-2020 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI Modules OpenMP(3.1) 
Delegates (built-in): bzlib fftw fontconfig freetype heic jng jp2 jpeg lcms ltdl lzma png tiff webp wmf x xml zlib

**Comment by snibgo** [comment]
I suppose you mean the filesize increases.

IM is a raster image processor. It reads and decompresses images into memory, then compresses and writes output files. If you don't actually want any image processing, but just want to change metadata, I suggest using exiftool instead.

**Reply by Jmuccigr** [reply]
Yes, in that particular example, that would work, but I'm also doing other stuff with the files and getting these size increases. So is there a way to tell what settings the file is already using, if that's an applicable idea here? TIA.

**Reply by fmw42** [reply]
What other processing?  Where did the original image come from? Did it use some other quantization tables that might be efficient?  Was it a graphic image that has few colors? Doing processing might increase the number of colors and thus the file size.

**Reply by Jmuccigr** [reply]
Happy to share the image. It's a scan of a book page. So it's in color, but mainly sepia-toned old pages. sRGB. Fairly large: 3344x4898.

Other processing? Well, some trimming, maybe some added borders to get them all to a similar size.

**Reply by fmw42** [reply]
Post the image

**Comment by Jmuccigr** [comment]
Zipped because Github doesn't like jp2.

[lesedificesantiq00desg_0005.jp2.zip](https://github.com/ImageMagick/ImageMagick/files/5220161/lesedificesantiq00desg_0005.jp2.zip)

**Comment by snibgo** [comment]
> So is there a way to tell what settings the file is already using, ...

exiftool doesn't show JP2 settings that correspond to the IM JP2 defines shown on https://imagemagick.org/script/defines.php. That make me doubt that they are recorded in the file.

**Reply by Jmuccigr** [reply]
Yeah, I had noticed that, but was hoping...
